common/crypto/openssl/rsa.c

```diff
@@ -121,6 +159,73 @@ static oe_result_t _get_public_key_get_modulus_or_exponent(
 
     return result;
 }
+#else
+static oe_result_t _get_public_key_get_modulus_or_exponent(
+    const oe_public_key_t* public_key,
+    uint8_t* buffer,
+    size_t* buffer_size,
+    bool get_modulus)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    size_t required_size = 0;
+    const BIGNUM* bn = NULL;
+    BIGNUM* e = NULL;
+    BIGNUM* n = NULL;
+
+    /* Check for invalid parameters */
+    if (!public_key || !buffer_size)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* If buffer is null, then buffer_size must be zero */
+    if (!buffer && *buffer_size != 0)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* Select modulus or exponent */
+    EVP_PKEY_get_bn_param(public_key->pkey, OSSL_PKEY_PARAM_RSA_N, &n);
```

Comment:
Can this function fail?

Reply:
Per https://manpages.debian.org/experimental/libssl-doc/EVP_PKEY_get_bn_param.3ssl.en.html, this function can fail. Please check the return value.